core/node/libp2p/rcmgr_defaults.go

```diff
@@ -191,9 +130,9 @@ Run 'ipfs swarm limit all' to see the resulting limits.
 		}
 
 		// Scale System.StreamsInbound as well, but use the existing ratio of StreamsInbound to ConnsInbound
-		defaultLimitConfig.System.StreamsInbound = rcmgr.LimitVal(maxInboundConns * int64(defaultLimitConfig.System.StreamsInbound) / int64(defaultLimitConfig.System.ConnsInbound))
-		defaultLimitConfig.System.ConnsInbound = rcmgr.LimitVal(maxInboundConns)
+		partialLimits.System.StreamsInbound = rcmgr.LimitVal(maxInboundConns * int64(partialLimits.System.StreamsInbound) / int64(partialLimits.System.ConnsInbound))
+		partialLimits.System.ConnsInbound = rcmgr.LimitVal(maxInboundConns)
 	}
 
-	return defaultLimitConfig.Build(orig), nil
+	return partialLimits.Build(rcmgr.DefaultLimits.Scale(int64(maxMemory), maxFD)), msg, nil
```

Comment:
There are a few important logic changes here and I'm not sure they're right...

1. Previously we were doing the autoscaling before the ``if cfg.ConnMgr.Type.WithDefault(config.DefaultConnMgrType) != "none"`` code that ensures ``System.ConnsInbound`` is high enough.  That was giving us the logic of ``max(computed System.ConnsInbound, connmgrHighWaterTimesTwo)``.  It now seems like we just get ``connmgrHighWaterTimesTwo``.
2. Prevously we were scaling some base limits by maxMemroy and maxFD.  Now we're scaling rcmgr.DefaultLimits.  I'm confused why ``rcmgr.DefaultLimits`` is even in the picture.  ``partialLimits`` already has all the scopes defined.  Why do we need DefaultLimits.

Reply:
1. this is a mistake that slipped through while rebasing and fixing conflicts, fixed.
2. partial limit is not scalled, we used to define both a base and a scalling config, Antonio changed it to now where we only define a base and then use default scalling, I don't know what are the impacts of not using a custom scalling.